src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -320,9 +320,9 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
                         continue 2;
                     }
 
-                    if ($attribute instanceof AutowireCallable) {
-                        $value = $attribute->buildDefinition($value, $type, $parameter);
-                        $value = $this->doProcessValue($value);
+                    if ($attribute instanceof AutowireInline) {
+                        $value = $this->container->getDefinition(ContainerBuilder::hash($attribute));
+                        $value = $this->processValue($value);
```

Comment:
```suggestion
                        $value = $attribute->buildDefinition($value, $type, $parameter);
                        $value = new Reference('.autowire_inline.'.ContainerBuilder::hash($value));
```

Reply:
instead of building a definition again here, shouldn't `ResolveAutowireInlineAttributesPass` inject a reference to the definition it created in the arguments (or maybe a binding depending on what works best) ?

Reply:
yes, that's what my code suggests, but first we need to build the definition so that we can get its hash

Reply:
Done [here](https://github.com/symfony/symfony/compare/dd16390a5290a9a08b2c997fa9d959296bb6f0e0..c94322321acafdbaf32044e58b213223645497ce)